agent/recorder-agent.c

```diff
@@ -0,0 +1,419 @@
+      cmd);
+
+  hwangsae_recorder_agent_start_recording (self, arg_id);
+
+  hwangsae1_dbus_recorder_interface_complete_start (object, invocation);
+
+  return TRUE;
+}
+
+gboolean
+    hwangsae_recorder_agent_recorder_interface_handle_stop
+    (Hwangsae1DBusRecorderInterface * object,
+    GDBusMethodInvocation * invocation, gchar * arg_id, gpointer user_data) {
+  g_autofree gchar *cmd = NULL;
+  g_autofree gchar *response = NULL;
+  GError *error = NULL;
+  HwangsaeRecorderAgent *self = (HwangsaeRecorderAgent *) user_data;
+
+  g_debug ("hwangsae_recorder_agent_recorder_interface_handle_stop, cmd %s",
+      cmd);
+
+  hwangsae_recorder_agent_stop_recording (self, arg_id);
+
+  hwangsae1_dbus_recorder_interface_complete_stop (object, invocation);
+
+  return TRUE;
+}
+
+static void
+hwangsae_recorder_agent_dispose (GObject * object)
+{
+  HwangsaeRecorderAgent *self = HWANGSAE_RECORDER_AGENT (object);
+
+  g_clear_object (&self->recorder);
+  g_clear_object (&self->settings);
+  g_clear_object (&self->manager);
+  g_clear_object (&self->recorder_interface);
+
+  G_OBJECT_CLASS (hwangsae_recorder_agent_parent_class)->dispose (object);
+}
+
+static void
+hwangsae_recorder_agent_class_init (HwangsaeRecorderAgentClass * klass)
+{
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+  GApplicationClass *app_class = G_APPLICATION_CLASS (klass);
+
+  gobject_class->dispose = hwangsae_recorder_agent_dispose;
+
+  app_class->dbus_register = hwangsae_recorder_agent_dbus_register;
+  app_class->dbus_unregister = hwangsae_recorder_agent_dbus_unregister;
+}
+
+static gboolean
+signal_handler (GApplication * app)
+{
+  g_application_release (app);
+
+  return G_SOURCE_REMOVE;
+}
+
+static void
+hwangsae_recorder_agent_init (HwangsaeRecorderAgent * self)
+{
+  gchar *uid = NULL;
+
+  self->is_recording = FALSE;
+  self->got_file_created_signal = FALSE;
+  self->got_file_completed_signal = FALSE;
+
+  self->settings = g_settings_new (HWANGSAE_RECORDER_SCHEMA_ID);
+
+  self->recorder = hwangsae_recorder_new ();
+
+  self->manager = hwangsae1_dbus_manager_skeleton_new ();
+
+  hwangsae1_dbus_manager_set_status (self->manager, 1);
+
+  self->recorder_interface = hwangsae1_dbus_recorder_interface_skeleton_new ();
+
+  g_signal_connect (self->recorder_interface, "handle-start",
+      G_CALLBACK (hwangsae_recorder_agent_recorder_interface_handle_start),
+      self);
+
+  g_signal_connect (self->recorder_interface, "handle-stop",
+      G_CALLBACK (hwangsae_recorder_agent_recorder_interface_handle_stop),
+      self);
+
+  hwangsae_recorder_set_container (self->recorder, HWANGSAE_CONTAINER_TS);
+  g_signal_connect (self->recorder, "stream-connected",
+      (GCallback) stream_connected_cb, self);
+  g_signal_connect (self->recorder, "file-created",
+      (GCallback) file_created_cb, self);
+  g_signal_connect (self->recorder, "file-completed",
+      (GCallback) file_completed_cb, self);
```

Comment:
I guess you copied the initial version from the recorder test, right? I think the agent shouldn't need  to connect `"file-*"` signals. You should then also remove any functions and variables referenced solely from `file_*_cb`.

Reply:
Yes, I copied those functions and let them to perform some sanity checks. I now think that at least we should get a `file-created` callback to get the record-id to be used. Do you agree?

Reply:
Yes, such use of the signal makes sense.

Reply:
I still don't see file_created_cb, file_completed_cb and stream_disconnected_cb doing anything meaningful, so we better remove them. There are also some leftover variables like got_file_completed_signal.